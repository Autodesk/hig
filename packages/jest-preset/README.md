# @hig/jest-preset

Jest preset for HIG packages.

## Usage

_Example `package.json`:_

```json
{
  "jest": {
    "preset": "@hig/jest-preset"
  }
}
```
