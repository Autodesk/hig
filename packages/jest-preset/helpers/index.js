/* eslint-disable global-require */
module.exports = {
  takeSnapshotsOf: require("./takeSnapshotsOf")
};
