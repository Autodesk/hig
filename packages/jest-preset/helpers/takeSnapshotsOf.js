const React = require("react");
const renderer = require("react-test-renderer");

module.exports = function takeSnapshotsOf(Component, cases, rendererOptions) {
  cases.forEach(({ desc, props }) => {
    it(desc, () => {
      const vDom = React.createElement(Component, props);
      const tree = renderer.create(vDom, rendererOptions).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
};
