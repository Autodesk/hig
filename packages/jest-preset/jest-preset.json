{
  "collectCoverageFrom": ["**/src/**/*.{js,jsx}"],
  "coverageDirectory": "<rootDir>/coverage",
  "coveragePathIgnorePatterns": [
    "__gemini__",
    "__stories__",
    "docs",
    "examples/redux",
    "node_modules",
    "packages/jest-preset",
    "packages/vanilla",
    "src/playground"
  ],
  "coverageThreshold": {
    "global": {
      "branches": 50,
      "functions": 50,
      "lines": 50,
      "statements": 50
    }
  },
  "moduleNameMapper": {
    "\\.(css|scss|svg)$": "@hig/jest-preset/fileMock.js"
  },
  "modulePaths": ["node_modules"],
  "moduleFileExtensions": ["js", "jsx", "json"],
  "setupFiles": ["raf/polyfill"],
  "setupTestFrameworkScriptFile": "@hig/jest-preset/setupTests.js",
  "testPathIgnorePatterns": [
    "docs",
    "packages/babel-preset",
    "packages/jest-preset",
    "packages/vanilla"
  ],
  "unmockedModulePathPatterns": ["node_modules"]
}
