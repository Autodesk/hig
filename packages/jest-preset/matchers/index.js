require("./toBeConstants");
require("./toImplementHIGInterfaceOf");
require("./toHavePropTypesAndDocgenInfo");
