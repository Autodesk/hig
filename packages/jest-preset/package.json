{
  "private": true,
  "name": "@hig/jest-preset",
  "version": "0.1.0",
  "devDependencies": {
    "babel-core": "^6.26.3",
    "babel-jest": "^23.4.0",
    "enzyme": "^2.9.1",
    "jest": "^21.2.1",
    "jest-enzyme": "^3.8.3",
    "jest-extended": "^0.6.0",
    "raf": "^3.4.0",
    "react-test-renderer": "^15.6.1"
  },
  "eslintConfig": {
    "extends": "@hig"
  },
  "jest": {
    "preset": "@hig/jest-preset"
  }
}
