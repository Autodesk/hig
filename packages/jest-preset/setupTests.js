require("jest-enzyme");
require("jest-extended");
require("./matchers");
