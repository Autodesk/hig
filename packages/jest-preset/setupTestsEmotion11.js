require("jest-extended");
const createSerializer = require("@emotion/jest").createSerializer;
const emotion = require("emotion");
const enzyme = require("enzyme");
const Adapter = require("@wojtekmaj/enzyme-adapter-react-17");

enzyme.configure({ adapter: new Adapter() });

require("./matchers");
require("./utils");

expect.addSnapshotSerializer(createSerializer());
