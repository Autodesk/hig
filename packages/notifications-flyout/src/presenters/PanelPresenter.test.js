import { ENTERED } from "react-transition-group/Transition";
import { takeSnapshotsOf } from "@hig/jest-preset/helpers";

const PanelPresenter = require("./PanelPresenter").default;

describe("notifications-flyout/presenters/PanelPresenter", () => {
  takeSnapshotsOf(PanelPresenter, [
    {
      desc: "renders without props",
      props: {
        innerRef: () => {},
      },
    },
    {
      desc: "renders with all props",
      props: {
        children: "foobar",
        heading: "Heading",
        innerRef: () => {},
        listMaxHeight: "3000px",
        loadingTransitionState: ENTERED,
        onScroll: () => {},
        refListWrapper: () => {},
      },
    },
  ]);
});
