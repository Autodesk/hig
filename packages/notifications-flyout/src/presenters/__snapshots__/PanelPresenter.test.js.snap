// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notifications-flyout/presenters/PanelPresenter renders with all props 1`] = `
.emotion-1 {
  width: 300px;
  overflow-y: auto;
  overflow-x: hidden;
}

.emotion-4 {
  background-color: #ffffff;
  border-radius: 4px;
  border: none;
  box-shadow: 0 0 16px rgba(0,0,0,0.2);
}

.emotion-3 {
  position: relative;
}

.emotion-0 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.428571429;
  margin: 0;
  text-align: initial;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 44px;
  overflow: hidden;
  background-color: transparent;
  opacity: 1;
  -webkit-transition-property: height,opacity;
  transition-property: height,opacity;
  -webkit-transition-duration: 300ms;
  transition-duration: 300ms;
  -webkit-transition-timing-function: ease;
  transition-timing-function: ease;
  pointer-events: none;
}

<div
  className="emotion-4"
  style={
    Object {
      "maxHeight": undefined,
    }
  }
>
  <div
    className="emotion-3"
  >
    <header
      className="emotion-0"
      style={
        Object {
          "borderBottom": "1px solid rgba(60, 60, 60, 0.25)",
          "padding": "12px
        24px",
        }
      }
    >
      Heading
    </header>
    <section
      className="emotion-1"
      style={
        Object {
          "maxHeight": "3000px",
        }
      }
    >
      <div
        onScroll={[Function]}
        role="list"
      >
        foobar
      </div>
    </section>
    <footer
      className="emotion-2"
    >
      <div
        data-props="{
  \\"size\\": \\"s\\"
}"
      />
    </footer>
  </div>
</div>
`;

exports[`notifications-flyout/presenters/PanelPresenter renders without props 1`] = `
.emotion-1 {
  width: 300px;
  overflow-y: auto;
  overflow-x: hidden;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 0;
  overflow: hidden;
  background-color: transparent;
  opacity: 0;
  -webkit-transition-property: height,opacity;
  transition-property: height,opacity;
  -webkit-transition-duration: 300ms;
  transition-duration: 300ms;
  -webkit-transition-timing-function: ease;
  transition-timing-function: ease;
  pointer-events: none;
}

.emotion-4 {
  background-color: #ffffff;
  border-radius: 4px;
  border: none;
  box-shadow: 0 0 16px rgba(0,0,0,0.2);
}

.emotion-3 {
  position: relative;
}

.emotion-0 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.428571429;
  margin: 0;
  text-align: initial;
}

<div
  className="emotion-4"
  style={
    Object {
      "maxHeight": undefined,
    }
  }
>
  <div
    className="emotion-3"
  >
    <header
      className="emotion-0"
      style={
        Object {
          "borderBottom": "1px solid rgba(60, 60, 60, 0.25)",
          "padding": "12px
        24px",
        }
      }
    >
      Notifications
    </header>
    <section
      className="emotion-1"
      style={
        Object {
          "maxHeight": undefined,
        }
      }
    >
      <div
        onScroll={undefined}
        role="list"
      />
    </section>
    <footer
      className="emotion-2"
    >
      <div
        data-props="{
  \\"size\\": \\"s\\"
}"
      />
    </footer>
  </div>
</div>
`;
